server/api/collections.js

```diff
@@ -454,59 +455,64 @@ router.post("collections.export", auth(), async (ctx) => {
   ctx.assertUuid(id, "id is required");
 
   const user = ctx.state.user;
+  const team = await Team.findByPk(user.teamId);
+  authorize(user, "export", team);
+
   const collection = await Collection.scope({
     method: ["withMembership", user.id],
   }).findByPk(id);
-  authorize(user, "export", collection);
 
-  const filePath = await archiveCollection(collection);
+  ctx.assertPresent(collection, "Collection should be present");
+  authorize(user, "read", collection);
 
-  await Event.create({
-    name: "collections.export",
-    collectionId: collection.id,
-    teamId: user.teamId,
-    actorId: user.id,
-    data: { title: collection.title },
-    ip: ctx.request.ip,
+  const name = collection ? collection.name : team.name;
+  const acl = process.env.AWS_S3_ACL || "private";
+  const key = getAWSKeyForFileOp(team.id, name, acl);
+
+  let exportData;
+  exportData = await FileOperation.create({
+    type: "export",
+    state: "creating",
+    key,
+    url: null,
+    size: 0,
+    collectionId: id,
+    userId: user.id,
+    teamId: team.id,
+  });
+
+  exportCollections(user.teamId, user.id, user.email, id, exportData.id);
+
+  exportData = await FileOperation.findByPk(exportData.id, {
+    include: [
+      {
+        model: User,
+        as: "user",
+      },
+      {
+        model: Collection,
+        as: "collection",
+      },
+    ],
   });
 
-  ctx.attachment(`${collection.name}.zip`);
-  ctx.set("Content-Type", "application/force-download");
-  ctx.body = fs.createReadStream(filePath);
+  ctx.body = {
+    success: true,
+    data: { fileOperation: presentFileOperation(exportData) },
```

Comment:
Using fileOperation as key instead of not presenting it directly because the response is from a collection endpoint, and fileOperation is different model.